uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrix.php

```diff
@@ -151,9 +151,7 @@ private function getStatusMatrixForCurrentUser($leaveRequestContactID) {
 
     if ($currentUserID == $leaveRequestContactID) {
       $statusMatrix = $this->getStaffStatusMatrix();
-    }
-
-    if ($this->shouldUseManagerMatrixForCurrentUser($leaveRequestContactID)) {
+    } elseif ($this->shouldUseManagerMatrixForCurrentUser($leaveRequestContactID)) {
```

Comment:
just about the logic here, if `($currentUserID == $leaveRequestContactID)` the fix was to change this to else if because `shouldUseManagerMatrixForCurrentUser` would return true. But does it make sense to change the logic in `shouldUseManagerMatrixForCurrentUser` to return false in this case since that's what seems to be the root cause?